File: java/src/main/java/com/datafiniti/importer/Baseline.java
package com.datafiniti.importer;

import redis.clients.jedis.Jedis;
import java.io.IOException;

class Baseline {
    public static void run() throws IOException {
        ESClient.connect();
        ESClient.setup();

        seedRedis(10_000);

        importRecords();
    }

    public static void importRecords() throws IOException {
        Jedis jedis = new Jedis("redis");

        System.out.println("importing records from redis -> elasticsearch");

        String record;
        while ((record = jedis.rpop("records")) != null) {
            ESClient.insert(record);
        }

        System.out.println("done importing.");
    }

    public static void seedRedis(Integer numRecords) throws IOException {
        Jedis jedis = new Jedis("redis");

        System.out.println("flushing redis");
        jedis.flushAll();

        System.out.println("seeding redis with " + numRecords + " records.");
        for (int i = 0; i < numRecords; i += 1) {
            jedis.rpush("records", Utils.createRecord());

            if (i % (numRecords / 10) == 0) {
                System.out.println(i + " records seeded.");
            }
        }

        jedis.close();
    }
}

File: nodejs/baseline.js
const Redis = require('ioredis');
const elasticsearch = require('./clients/elasticsearch');

const { createRecord, setupElasticsearch } = require('./utils');

(async () => {
    await setupElasticsearch();
    await seedRedis(10000);
    await importRecords();
})();

async function importRecords() {
    console.log('importing records from redis -> elasticsearch');
    
    const redis = new Redis({ host: 'redis' });
    let record;

    do {
        record = await redis.rpop('records');
        record && await elasticsearch.index({ index: 'records', type: 'all', body: record });
    } while (record);

    await redis.disconnect();
}

async function seedRedis(numRecords) {
    const redis = new Redis({ host: 'redis' });
    
    console.log('flushing redis');
    await redis.flushall();
    
    for (let i = 0; i < numRecords; i += 1) {
    
        await redis.rpush('records', JSON.stringify(createRecord()));

        if (i % (numRecords / 10) === 0) {
            console.log(`${i} records seeded in redis`);
        }
    }

    console.log('done seeding redis');
    await redis.disconnect();
}